compiler/modules/print.wasm/relooper.bal

```diff
@@ -1,37 +1,482 @@
+                            ifBlock.name = label;
+                            foreach Expression expr in ifCode {
+                               ifBlock.body.push(expr);                            
+                            }
+                        }
+                        ifBlock.body.push(ifBreakBlock);
+                        ifBody = ifBlock;
+                        isCond = false;
+                    }
+                    else if handledBlock != () {
+                        WasmBlock elseBlock = {};
+                        Break br = { label : label };
+                        if multiple.handledBlocks.keys().length() == 2 {
+                            Expression[] elseCode = self.renderShape(handledBlock);
+                            foreach Expression expr in elseCode {
+                                elseBlock.body.push(expr);                            
+                            }
+                            WasmBlock elseBreak = {
+                                body: [br]
+                            };
+                            elseBlock.body.push(elseBreak);   
+                        }
+                        else {
+                            elseBlock.body.push(br);
+                        }
+                        elseBody = elseBlock;
+                    }
+                }
+                if condition != () && ifBody != () {
+                    If ifExpr = {
+                        condition : condition,
+                        ifBody : ifBody,
+                        elseBody : elseBody
+                    };
+                    innerBlock.body.push(ifExpr);
+                }
+            }
+            else {
+                io:println("unimplemented");
+            }
+            if next != () {
+                Expression[] nextExprs = self.renderShape(next);
+                if nextExprs.length() == 1 {
+                    children.push(nextExprs[0]);
+                }
+                else{
+                    foreach Expression expr in  nextExprs{
+                        nextBlock.body.push(expr);
+                    }
+                    children.push(nextBlock);
+                }
+            }
+           
+        }
+        else {
+            innerBlock.body.push(simple.inner.code);
+        }
+        return children;
+    }
+
+    function renderLoopShape(LoopShape loop) returns Expression[] {
+        panic error("unimplemented");
+
+    }
+
+    function renderMultipleShape(MultipleShape multiple) returns Expression[] {
+        panic error("unimplemented");
+    }
+
+    function printSpaces(int spaceCount) returns string {
+        string spaces = "";
+        foreach int i in 0...spaceCount {
+            spaces += " ";
+        }
+        return spaces;
     }
 
-    function calculate(Block entry) {
-        panic error("unimplemented");        
+    function makeBlockText(Expression[] result, int spacesCount) returns string {
```

Comment:
Push each string part to an array and join them to get the full string, search `"".'join(...` in the code base to see examples.

Reply:
fixed